server/data/courseClient.test.ts

```diff
@@ -67,6 +68,68 @@ pactWith({ consumer: 'Accredited Programmes UI', provider: 'Accredited Programme
     })
   })
 
+  describe('createParticipation', () => {
+    describe('when `courseId` is provided', () => {
+      const courseParticipation = courseParticipationFactory.withCourseId().build()
+      const { courseId, prisonNumber } = courseParticipation
+
+      beforeEach(() => {
+        provider.addInteraction({
+          state: `A person exists with prison number ${prisonNumber}`,
+          uponReceiving: `A request to create a course participation for person "${prisonNumber}"`,
+          willRespondWith: {
+            body: Matchers.like(courseParticipation),
+            status: 201,
+          },
+          withRequest: {
+            body: { courseId, prisonNumber } as CourseParticipation,
```

Comment:
Ahhh I see what's going on with the casting here. It looks like the problem is that `courseId` could be `undefined`, which would make it invalid JSON - the `body` here needs to be valid JSON. This isn't a `CourseParticipation` though - that would require `id`, `addedBy` and `createdAt` at a minimum - so it might be better to either cast `courseId` as a `string`, or update L73 to:

```ts
      const courseParticipation = courseParticipationFactory.withCourseId().build() as CourseParticipation & {
        courseId: string
      }
```

That would narrow out the possibility of it being `undefined`

Same in the test below